src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -1725,31 +1733,22 @@ class GracefulGoaway : public grpc_core::RefCounted<GracefulGoaway> {
 
   static void OnPingAckLocked(void* arg, grpc_error_handle /* error */) {
     auto* self = static_cast<GracefulGoaway*>(arg);
-    grpc_timer_cancel(&self->timer_);
+    self->t_->event_engine->Cancel(
+        std::exchange(self->timer_handle_, TaskHandle::kInvalid));
     self->MaybeSendFinalGoawayLocked();
     self->Unref();
   }
 
-  static void OnTimer(void* arg, grpc_error_handle error) {
-    auto* self = static_cast<GracefulGoaway*>(arg);
-    if (!error.ok()) {
-      self->Unref();
-      return;
-    }
-    self->t_->combiner->Run(
-        GRPC_CLOSURE_INIT(&self->on_timer_, OnTimerLocked, self, nullptr),
-        absl::OkStatus());
-  }
-
   static void OnTimerLocked(void* arg, grpc_error_handle /* error */) {
     auto* self = static_cast<GracefulGoaway*>(arg);
+    self->timer_handle_ = TaskHandle::kInvalid;
```

Comment:
what's the significance of setting `timer_handle_` to `kInvalid`? Is it just a sanity thing? If so, can you add a quick comment please?